File: apps/qizhi/mobile/MentorAI-iOS/MentorAI/User/UserAPI.swift
import Foundation

struct UserAPI {
    let client: APIClient

    init(baseURL: URL) {
        self.client = URLSessionAPIClient(baseURL: baseURL)
    }

    init(client: APIClient) {
        self.client = client
    }

    func currentUser(token: String) async throws -> UserProfile {
        try await client.get("/user/current", query: [], bearerToken: token)
    }
}

File: qizhi/mobile/MentorAI-Android/app/src/main/kotlin/com/mentorai/app/user/UserAPI.kt
package com.mentorai.app.user

import com.mentorai.app.networking.APIClient

class UserAPI(private val client: APIClient) {
    /** GET /user/current — returns the currently authenticated user (matches the iOS path). */
    suspend fun current(token: String): UserProfile =
        client.getEnvelope("/user/current", bearerToken = token)
}
